js/app.js

```diff
@@ -26,7 +27,7 @@ function initiateEventListeners() {
 
 async function loadBlogItems() {
     var btnLoadMore = document.getElementById("loadMoreBlogItems");
-    const api = btnLoadMore.dataset.next_page || buildBlogItemsUrl(tag);
+    const api = btnLoadMore.dataset.next_page || buildBlogItemsUrl(tags);
```

Comment:
Hmm, I had to think for a bit - it is because on the first load of the page, next url is set further down - line 43:

`setNextBatchLoad(json.pagination.next_page);`

Where as the selected tags are passed in query string and should be processed the on the first call of loadBlogItems(). If next url is defined, this already has the selected tags so no need to process the tags again...

Subsequent calls of loadBlogItems() don't re-load the page, just load the next batch...

Reply:
hmm. I still can't follow this logic. I must try it out i think

Reply:
Well, if you cannot follow it there must be something wrong with it...

Reply:
I'll play with this after you merge and try to understand it.